### PR TITLE
fix(meta): apply snapshot will block until newest mp snapshot persist…

### LINKED ISSUE #2230: [Bug]: metanode create metapartition raft panic.
**Describe the bug**
```
dTo]applied(96112591) is out of range [prevApplied(105291801), committed(105292441)])
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/depends/tiglabs/raft/util/runtime.go:38
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/depends/tiglabs/raft/util/runtime.go:28
/usr/local/go/src/runtime/panic.go:965
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/depends/tiglabs/raft/raft_log.go:235
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/depends/tiglabs/raft/raft_fsm.go:123
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/depends/tiglabs/raft/raft.go:135
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/depends/tiglabs/raft/server.go:162
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/raftstore/raftstore.go:192
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/metanode/partition.go:440
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/metanode/partition.go:342
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/metanode/partition.go:297
/root/tmp/cfs/go/src/github.com/cubefs/cubefs/metanode/manager.go:393
```
when metapartition recover from leader's snapshot, and then the meta node killed before store newest snapshot file to disk, mp create raft will be panic when start metanode again.

**To Reproduce**
1. when store snapshot to disk, update code to sleep 1 minutes, to suppose that slow store snapshot to disk case.
2. update retainlogs count to 20, so snapshot step can  be happened easily.
3. try to create 100 file with a new volume A in the root dir.
4. stop the A's first mp metapartition replica B
5. create another 100 files, sleep 5 minutes, create 100 files, sleep 5 miutes again
6. then start replica B，the replica can't start and will panic.

**Expected behavior**
metanode service can start normally.

### PR DESCRIPTION
**Pull request by Victor1319**
… to disk success

Signed-off-by: Victor1319 <834863182@qq.com>

Closes #2230

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: metanode/partition_fsm.go [hunk 1]
```diff
@@ -439,6 +439,21 @@ func (mp *metaPartition) ApplySnapshot(peers []raftproto.Peer, iter raftproto.Sn
...
 		txRbDentryTree = NewBtree()
 	)
+
+	blockUntilStoreSnapshot := func() {
+		ticker := time.NewTicker(5 * time.Second)
+		defer ticker.Stop()
+
+		log.LogWarnf("ApplySnapshot: start to block until store snapshot to disk, mp %d, appid %d", mp.config.PartitionId, appIndexID)
+
+		for range ticker.C {
```
Comment (true1064) [inline metanode/partition_fsm.go:449]: need consider the senario that mp.stopC received msg while waiting?

File: metanode/partition.go [hunk 2]
```diff
@@ -980,6 +982,11 @@ func (mp *metaPartition) store(sm *storeMsg) (err error) {
 		return
 	}
 	err = os.RemoveAll(backupDir)
+	if err != nil {
+		return
+	}
+
+	mp.storedApplyId = sm.applyIndex
```
Comment (true1064) [inline metanode/partition.go:989]: mp.storedApplyId should use atomic operation

**Review by true1064 — CHANGES_REQUESTED** [review]

**Review by true1064 — APPROVED** [review]

**Comment by true1064** [comment]
LGTM

**Review by xiangcai1215 — APPROVED** [review]

**Review by xiangcai1215 — APPROVED** [review]

**Review by leonrayang — APPROVED** [review]
LGTM

**Pull request merged**